### PR TITLE
Added examples which use total ordering

### PR DESCRIPTION
**Pull request by SamArch27**
I'm adding programs that require a total ordering over the input to compute a solution since we can speed them up dramatically with the new auto-indexing technique.

So far, I've added:
- Minimum Spanning Tree (Prim's Algorithm)
- Convex Hull (Monotone Chain Algorithm)
- Fractional Knapsack (Greedy Algorithm)
- Greedy Coloring (Welsh Powell Algorithm)

TODO: Fix CMAKE/Automake issues.

I might add some more examples of algorithms with total orderings later if needed.

### CODE REVIEW

**Comment by b-scholz** [comment]
Please can you add the examples to CMakelist.txt as well?

Unfortunately, we have to maintain cmake and autoconf/automake simultaneously until the transition to cmake has been fully completed.

**Comment by b-scholz** [comment]
Also the blanks at the end of the line must be deleted to pass the style checker...

**Comment by SamArch27** [comment]
> Please can you add the examples to CMakelist.txt as well?
> 
> Unfortunately, we have to maintain cmake and autoconf/automake simultaneously until the transition to cmake has been fully completed.

As far as I'm aware I've added to both automake and cmake. I added to the CMakeLists.txt in /examples/ as well as to examples.at

**Comment by b-scholz** [comment]
There is an example for external functors in tests/interface/functors. 
You need to build a functor library in the directory. The build differs for cmake and autoconf/automake.

File: tests/example/CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,110 @@ function(NEGATIVE_TEST NAME)
     souffle_negative_test(${NAME} example)
 endfunction()
 
+# Make sure that functors are built for the graph_coloring example
```
Comment (b-scholz) [inline tests/example/CMakeLists.txt:17]: I don't think you need to copy this. Why can you not imitate the functors example.

Comment (b-scholz) [reply tests/example/CMakeLists.txt:17]: in tests/interface/CMakelist.txt just add 
```souffle_positive_functor_test(graph coloring CATEGORY interface)```

**Review by b-scholz — APPROVED** [review]

**Pull request merged**